### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by ctrlaltelite2012**
Please review and merge.

### CODE REVIEW

File: NCM_APIv2_class/v2_example_class.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+                                                  503,  # 503 Service Unavailable
+                                                  504,  # 504 Gateway Timeout
+                                                  ],
+                                )
+                a = requests.adapters.HTTPAdapter(max_retries=retries)
+                s.mount('https://', a)
+
+                try:
+                    logging.info(url)
+                    print(url)
```
Comment (gmcnutt) [inline NCM_APIv2_class/v2_example_class.py]: You don't need print() when you are using logging (and logging is the correct thing to use in a module meant to be used by other programs).

File: NCM_APIv2_class/v2_example_class.py [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+                    r = s.get(url, headers=self.headers,
+                              timeout=30, stream=True)
+
+                    if r.status_code != 200:
+                        logging.error(str(r.status_code) + ": " + str(r.text))
+                        s.get('https://accounts.cradlepointecm.com/logout')
```
Comment (gmcnutt) [inline NCM_APIv2_class/v2_example_class.py]: Why are you calling logout?

File: NCM_APIv2_class/v2_example_class.py [hunk 3]
```diff
@@ -0,0 +1,137 @@
...
+
+                    else:
+                        self.resp = json.loads(r.content.decode("utf-8"))
+
+                        if len(self.resp['data']) < 1:
+                            logging.warning('No response data was returned.')
```
Comment (gmcnutt) [inline NCM_APIv2_class/v2_example_class.py]: Don't warn. This is normal in some cases.

File: NCM_APIv2_class/v2_example_class.py [hunk 4]
```diff
@@ -0,0 +1,137 @@
...
+                os.makedirs(dir_path)
+            except OSError:
+                print (f"Creation of the directory %s failed" % dir_path)
+            else:
+                print ("Successfully created the directory %s " % dir_path)
+        else:
+            print ("%s already exists" % dir_path)
+
+
+if __name__ == '__main__':
```
Comment (gmcnutt) [inline NCM_APIv2_class/v2_example_class.py]: Oh, wait, this is a program, not a module. Never mind what I said earlier. Put this under "examples" not "lib" and replace the get.py that is already there, or merge it's features into this one. And never mind what I said about "logging" vs "print", in a standalone program either is fine but pick one.

**Comment by ctrlaltelite2012** [comment]
Did the recommend changes and created a different pull request. Closing this one.

**Pull request closed without merging**